Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>


From libc-alpha-return-31199-listarch-libc-alpha=sources dot redhat dot com at sourceware dot org Tue Jul 03 19:54:27 2012
Return-Path: <libc-alpha-return-31199-listarch-libc-alpha=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-alpha at sources dot redhat dot com
Received: (qmail 20151 invoked by alias); 3 Jul 2012 19:54:25 -0000
Received: (qmail 20139 invoked by uid 22791); 3 Jul 2012 19:54:23 -0000
X-SWARE-Spam-Status: No, hits=-4.8 required=5.0
	tests=AWL,BAYES_00,KHOP_RCVD_UNTRUST,RCVD_IN_DNSWL_HI,RCVD_IN_HOSTKARMA_W,T_RP_MATCHES_RCVD
X-Spam-Check-By: sourceware.org
From: Mike Frysinger <vapier at gentoo dot org>
To: libc-alpha at sourceware dot org
Subject: [PATCH] sunrpc: fix rpc bootstrap builds
Date: Tue,  3 Jul 2012 15:54:11 -0400
Message-Id: <1341345251-31730-1-git-send-email-vapier@gentoo.org>
Mailing-List: contact libc-alpha-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <libc-alpha.sourceware.org>
List-Subscribe: <mailto:libc-alpha-subscribe at sourceware dot org>
List-Archive: <http://sourceware.org/ml/libc-alpha/>
List-Post: <mailto:libc-alpha at sourceware dot org>
List-Help: <mailto:libc-alpha-help at sourceware dot org>, <http://sourceware dot org/ml/#faqs>
Sender: libc-alpha-owner at sourceware dot org
Delivered-To: mailing list libc-alpha at sourceware dot org

If you build & install glibc w/rpc disabled, you no longer have headers in
/usr/include/rpc/ (this is expected).  But if you try to build glibc w/rpc
enabled, this gets into a bad state due to the new rpc helpers that get
cross-compiled:

$ make
...
x86_64-pc-linux-gnu-gcc -m32   -D_RPC_THREAD_SAFE_ -D_GNU_SOURCE -DIS_IN_build \
	-include $objdir/config.h rpc_clntout.c -o $objdir/sunrpc/cross-rpc_clntout.o \
	-MMD -MP -MF $objdir/sunrpc/cross-rpc_clntout.o.dt -MT $objdir/sunrpc/cross-rpc_clntout.o -c
rpc_clntout.c:34:23: fatal error: rpc/types.h: No such file or directory
compilation terminated.
make: *** [$objdir/sunrpc/cross-rpc_clntout.o] Error 1

Signed-off-by: Mike Frysinger <vapier@gentoo.org>

2012-07-03  Mike Frysinger  <vapier@gentoo.org>

	* sunrpc/rpc_clntout.c: Change <rpc/types.h> to "rpc/types.h".
---
 sunrpc/rpc_clntout.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: libc/sunrpc/rpc_clntout.c
===================================================================
--- libc.orig/sunrpc/rpc_clntout.c	2012-07-04 16:09:16.000000000 -0700
+++ libc/sunrpc/rpc_clntout.c	2012-07-04 18:04:34.498085075 -0700
@@ -31,7 +31,7 @@
  */
 #include <stdio.h>
 #include <string.h>
-#include <rpc/types.h>
+#include "rpc/types.h"
 #include "rpc_parse.h"
 #include "rpc_util.h"
 #include "proto.h"
